### DISCUSSION TITLE
Growing degree day option for growth stages

### DISCUSSION THREAD

**Discussion by kthorp** [kthorp/pyfao56#52] [Ideas]
Several folks want this feature included. We've had initial discussions on how to do it, but progress is complicated by their being multiple ways that growing degree days can be calculated. Waiting for an official recommendation from FAO on appropriate equations for growing degree day calculations and understanding it should come with the 2024 revision of FAO56.

**Comment by isukendall** [comment]
My assumption is that the revision of FAO-56 will give something like GDDini, GDDdev, GDDmid, and GDDend instead of day lengths Lini, Ldev, Lmid, and Lend for the time periods that determine the length of the trapezoidal curve. 

Somewhat related, in a seasonal water balance it can be helpful to note observed phenological growth stages, and/or predict future growth stages. I created my own spreadsheet model in the past to predict leaf growth, anthesis, and maturity growth stages for maize, as a function of observed and/or predicted GDDs. This sort of functionality could lead to deeper analysis, for example separation or categorization of crop growth into phenological stages rather than just the FAO-56 stages that define the curve. It can also be helpful to have a "predicted" anthesis date, have a good idea of vegetative stages for fractional cover development, maturity for harvesting, etc. Many of our irrigation experiment treatments are dependent upon growth stage: for example, we might be full irrigation up to full cover, limited irrigation until anthesis, full irrigation from anthesis to R3, and limited irrigation beyond.

Both of these ideas start with establishment of a GDD model, which would likely require user-defined upper and lower limits for calculation of GDDs.

**Comment by kthorp** [comment]
Lots of discussion on this topic at the May 2025 quarterly pyfao56 meeting.  Word is that the paper describing revisions to FAO56 is currently in press with an expected publication by end of year.  I expect the new guidance will need to provide 1) a standard method for computation of growing degree days and 2) recommended values for growing degree day accumulation to switch to the next growth stage.  Assuming this is the case, it will be easy to add the GDD method to the model.  I expect it will entail 1) new parameters in the Parameters class to specify the GDD values for growth stages (in lieu of day numbers, Lini, Ldev, Lmin, Lend), 2) a switch in the constructor of Model class to select between the DAP method versus GDD method, and 3) a new section of code in Model class to compute daily GDD from temperature, accumulate it, and specify appropriate Kc and Kcb for that time step. There was discussion that the FAO recommendation will likely be similar to what is already being used in Idaho, which may include more than four inflection points.
